state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -397,6 +443,12 @@ pub mod pallet {
 			amount: AssetAmount,
 			reason: DispatchError,
 		},
+		RefundEgressIgnored {
```

Comment:
It seems like we're duplicating events (*EgressIgnored, *EgressScheduled, ...) - probably a good candidate for the events refactor. (maybe we just need `EgressScheduled { swap_id, egress_id }` ? The context of the egress can be derived from the swap and egress asset)

Reply:
We have been going back and forth on this... In the end we decided (and put it in the design document) that it would be more convenient for the product to have separate events. I will let @niklasnatter make the final call.

Reply:
Sure. We can revisit this in the context of the events refactor. I feel like we need to be clearer about when the contextual information is or isn't necessary on an event (even when it should be clear from the linked egress/swap/request). I am in favour of a single consistent approach.

Reply:
Yes for now I would prefer having the `RefundEgressIgnored` because it is consistent. I agree that this is something to revisit when we are refactoring the events and come up with a consistent approach.

About the duplication; from my experience with indexing Ethereum events I would say its worth duplicating some data. Eg if I am interested in all the outflows from the Statechain, it would be nice if i dont have to index swap events to figure out the amounts.

Reply:
Yes, totally! This is why we need a consistent approach: when we schedule an egress, we should emit an event for this (we currently don't!).